packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -30,6 +30,22 @@ class Carousel extends Foundation<
 > {
     public static displayName: string = `${DisplayNamePrefix}Carousel`;
 
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: CarouselProps,
+        prevState: CarouselState
+    ): null | CarouselState {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
```

Comment:
is it ok to return empty state?

Reply:
This is based on the requirements of the lifecycle hook - either return a state object or null to not update. :)